app/models/repository_stock_value.rb

```diff
@@ -87,13 +87,13 @@ def self.import_from_text(text, attributes, _options = {})
     nil
   end
 
-  def update_stock_with_ledger!(amount, reference, comment)
+  def update_stock_with_ledger!(amount, comment)
     delta = amount.to_d - self.amount.to_d
     repository_ledger_records.create!(
       user: last_modified_by,
       amount: delta,
       balance: amount,
-      reference: reference,
+      reference: repository_cell.repository_column.repository,
```

Comment:
I think the reference is supposed to be either the repository (if change came from repository management), or a MyModuleRepositoryRow, if it was done via task stock consumption.

Reply:
is it possible to change stock from task?

Reply:
Looks like there's separate logic for that, the `deduct_stock_balance` callback method in `app/models/my_module_repository_row.rb`.
Maybe that logic should be unified to use this method, in which case the reference would need to go back in as an argument.